dev-lang/lua/lua54-5.4.0~beta.recipe

```diff
@@ -0,0 +1,110 @@
+SUMMARY="A powerful, fast, light-weight, embeddable scripting language"
+DESCRIPTION="Lua combines simple procedural syntax with powerful data \
+description constructs based on associative arrays and extensible semantics. \
+Lua is dynamically typed, runs by interpreting bytecode for a register-based \
+virtual machine, and has automatic memory management with incremental garbage \
+collection, making it ideal for configuration, scripting, and rapid \
+prototyping."
+HOMEPAGE="https://www.lua.org/"
+COPYRIGHT="1994-2019 Lua.org, PUC-Rio"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="http://www.lua.org/work/lua-${portVersion/\~beta/-beta}.tar.gz"
+CHECKSUM_SHA256="5eb2824bc08469be9d9282c7298f001830ea013179ad0ae8a50600332568ebb9"
+SOURCE_DIR="lua-${portVersion/\~beta/-beta}"
+PATCHES="lua-$portVersion.patchset"
+if [ $effectiveTargetArchitecture = x86_gcc2 ]; then
+	PATCHES+="
+		lua-$portVersion.gcc2.patchset
+	"
+fi
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86_gcc2 x86"
+
+libVersion="${portVersion%.*}"
+libVersionCompat="$libVersion compat >= $libVersion"
+luaInc="`echo ${libVersion} | tr -d '.'`"
+luaVer="`echo $portVersion | cut -d~ -f1`"
+
+PROVIDES="
+	lua54$secondaryArchSuffix = $portVersion
+	cmd:lua$secondaryArchSuffix = $portVersion
+	cmd:lua$libVersion$secondaryArchSuffix = $portVersion
+	cmd:luac$secondaryArchSuffix = $portVersion
+	cmd:luac$libVersion$secondaryArchSuffix = $portVersion
+	lib:liblua$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	lua54${secondaryArchSuffix}_devel = $portVersion
+	devel:liblua$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	lua54$secondaryArchSuffix == $portVersion base
+	"
+
```

Comment:
Shouldn't we add a CONFLICTS here (like in lua52)?

Reply:
Probably, if the link is named the same.

Reply:
How do I check? lua.pc is called the same in all versions?

Reply:
Well, lua$libVersion.pc is version related ... (but writes over the lua.pc file if you check in Terminal)

Reply:
added CONFLICTS